include/numerics/dense_matrix.h

```diff
@@ -422,6 +406,16 @@ class DenseMatrix : public DenseMatrixBase<T>
    * Requires PETSc >= 3.1 since this was the first version to provide
    * the LAPACKgelss_ wrapper.
    */
+
+   void svdinv(DenseMatrix<Number> & Inverse);
```

Comment:
You may also want to rethink this interface. If the pseudo-inverse will be computed many times, it will be more efficient to reuse the sigma, U, and VT arrays rather than creating and destroying them each time. Passing 4 arguments to the function might be a better API:
```
void svdinv(DenseMatrix<Number> & Inverse,
const DenseVector<Real> & sigma,
const DenseMatrix<Number> & U,
const DenseMatrix<Number> & VT);
```